libsql/turso with spring jpa ending up with : "Failed to execute isValid() for connection, configure connection test query (null)"

I am trying to setup libsql as drop in replacement for sqlite. Has anyone faced this with spring boot

I am having jpa database-platform: org.hibernate.community.dialect.SQLiteDialect

```
diff --git a/build.gradle.kts b/build.gradle.kts
index 0e8f176..ebe647d 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -88,7 +88,7 @@ dependencies {
         exclude(group = "org.springframework.boot", module = "spring-boot-starter-logging")
     }
 //    implementation("org.springframework.boot:spring-boot-starter-data-r2dbc")
-    implementation("org.xerial:sqlite-jdbc")
+    implementation("com.dbeaver.jdbc:com.dbeaver.jdbc.driver.libsql:1.0.2")

 
     implementation("org.springframework.boot:spring-boot-starter-web") {
diff --git a/src/main/resources/application.yml b/src/main/resources/application.yml
index e1bb2b6..dc51793 100644
--- a/src/main/resources/application.yml
+++ b/src/main/resources/application.yml
@@ -9,10 +9,11 @@ spring:
     main:
         lazy-initialization: true
     datasource:
-        driver-class-name: org.sqlite.JDBC
-        username: sa
-        password: sa
-        url: jdbc:sqlite:db.sqlite3
+        driver-class-name: com.dbeaver.jdbc.driver.libsql.LibSqlDriver
+        password: x.x.x-x # Trust me valid token
+        url: jdbc:dbeaver:libsql:https://x.turso.io # Trust me valid url
+        hikari:
+            isolate-internal-queries: false # Tried with true as well
     jpa:
         database-platform: org.hibernate.community.dialect.SQLiteDialect
         generate-ddl: true
```

Failed to execute isValid() for connection, configure connection test query (null)
SQL Error: 0, SQLState: null
null
HHH000342: Could not obtain connection to query metadata
unable to obtain isolated JDBC connection [null]
Caused by: java.sql.SQLFeatureNotSupportedException

--- a/com.dbeaver.jdbc.driver.libsql/src/main/java/com/dbeaver/jdbc/driver/libsql/LibSqlConnection.java
+++ b/com.dbeaver.jdbc.driver.libsql/src/main/java/com/dbeaver/jdbc/driver/libsql/LibSqlConnection.java
@@ -28,6 +28,10 @@ import java.sql.PreparedStatement;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 public class LibSqlConnection extends AbstractJdbcConnection {
 
@@ -40,6 +44,7 @@ public class LibSqlConnection extends AbstractJdbcConnection {
     @NotNull
     private final Map<String, Object> driverProperties;
     private LibSqlDatabaseMetaData databaseMetaData;
+    private volatile boolean closed = false;
 
     public LibSqlConnection(
         @NotNull LibSqlDriver driver,
@@ -108,12 +113,47 @@ public class LibSqlConnection extends AbstractJdbcConnection {
 
     @Override
     public void close() throws SQLException {
-        client.close();
+        if (!closed) {
+            client.close();
+            closed = true;
+        }
     }
 
     @Override
     public boolean isClosed() {
-        return false;
+        return closed;
+    }
+
+    @Override
+    public boolean isValid(int timeout) throws SQLException {
+        if (timeout < 0) {
+            throw new SQLException("Invalid timeout value: " + timeout);
+        }
+        if (closed) {
+            return false;
+        }
+        CompletableFuture<Boolean> ping = CompletableFuture.supplyAsync(() -> {
+            try {
+                LibSqlUtils.executeQuery(this, "SELECT 1");
+                return Boolean.TRUE;
+            } catch (Exception e) {
+                return Boolean.FALSE;
+            }
+        });
+        try {
+            if (timeout == 0) {
+                return ping.get();
+            }
+            return ping.get(timeout, TimeUnit.SECONDS);
+        } catch (TimeoutException e) {
+            ping.cancel(true);
+            return false;
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return false;
+        } catch (ExecutionException e) {
+            return false;
+        }
     }
 
     @Override

--- a/com.dbeaver.jdbc.driver.libsql/src/test/java/com/dbeaver/jdbc/upd/driver/test/LibSqlDriverTest.java
+++ b/com.dbeaver.jdbc.driver.libsql/src/test/java/com/dbeaver/jdbc/upd/driver/test/LibSqlDriverTest.java
@@ -35,6 +35,14 @@ public class LibSqlDriverTest {
                 System.out.println("Driver: " + metaData.getDriverName());
                 System.out.println("Database: " + metaData.getDatabaseProductName() + " " + metaData.getDatabaseProductVersion());
 
+                System.out.println("isValid(5) = " + connection.isValid(5));
+                try {
+                    connection.isValid(-1);
+                    System.out.println("BUG: negative timeout did not throw");
+                } catch (SQLException e) {
+                    System.out.println("isValid(-1) correctly threw: " + e.getMessage());
+                }
+
                 System.out.println("Query:");
                 try (Statement dbStat = connection.createStatement()) {
                     try (ResultSet dbResult = dbStat.executeQuery("select * from testme")) {
@@ -61,6 +69,8 @@ public class LibSqlDriverTest {
                     }
                 }
 
+                connection.close();
+                System.out.println("isValid(5) after close = " + connection.isValid(5));
             }
         } finally {
             System.out.println("Finished (" + (System.currentTimeMillis() - startTime) + "ms)");

